### DISCUSSION TITLE
Composite Relationships

### DISCUSSION THREAD

**Discussion by oroalej** [laravel/framework#33564] [General]
Hi all,

Thank you for maintaining and creating this framework. 
I just want to ask if Laravel will support composite relationship out of the box in the future?

**Comment by jmarcher** [comment]
What do you mean with composite relationships? Can you elaborate?

**Reply by oroalej** [reply]
Hi @jmarcher 

I mean composite eloquent relationship. 

`protected $primaryKey = [ 'someId',  'anotherId' ]`

for the relationship;

```
 public function phone()
 {
    return $this->hasOne(App::class, 'foreign_key', [ 'someId',  'anotherId' ]);
 }
```

**Comment by parallels999** [comment]
https://github.com/topclaudy/compoships

**Comment by denizgolbas** [comment]
There is no way to define composite key in eloquent model default relation methods. You need to use composite key packages. But you can define composite keys in  database schema.
